aws-android-sdk-cognitoidentityprovider/src/main/java/com/amazonaws/mobileconnectors/cognitoidentityprovider/CognitoUserPool.java

```diff
@@ -589,4 +609,11 @@ protected UserContextDataType getUserContextData(String userId) {
         }
         return contextData;
     }
+
+    /**
+     * Returns the {@link AmazonCognitoIdentityProvider} instance
+     */
+    public AmazonCognitoIdentityProvider getClientInstance() {
```

Comment:
This codebase has multiple nested layers of accretion, and there are lots of different clients. It'd probably be better to be explicit, here:

```suggestion
    public AmazonCognitoIdentityProvider getAmazonCognitoIdentityProvider() {
```